clang/test/CodeGenCXX/reference-field.cpp

```diff
@@ -1,8 +1,8 @@
 // RUN: %clang_cc1 -emit-llvm -triple %itanium_abi_triple -o - %s -O2 | FileCheck %s
 
-// Make sure the call to b() doesn't get optimized out.
+// Make sure the call to b() is eliminated.
 extern struct x {char& x,y;}y;
 int b();      
 int a() { if (!&y.x) b(); }
```

Comment:
Clang-tidy complains about this line: https://godbolt.org/z/4e55z7bdv
```
warning: reference cannot be bound to dereferenced null pointer in well-defined C++ code; pointer may be assumed to always convert to true [-Wundefined-bool-conversion]
```

Reply:
Looking at the original commit, maybe change this to `char *a() { return &y.x; }`.  Or just delete the test, since we have plenty of other test coverage for references.